overlord/snapstate/handlers.go

```diff
@@ -753,17 +755,53 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 		if err != nil {
 			return err
 		}
+
+		// Re-compute icon download filepath and URL if the result info has the
+		// necessary information
+		if result.SnapID != "" {
+			targetIconFn = backend.IconDownloadFilename(result.SnapID)
+		}
+		if resultIconURL := result.Media.IconURL(); resultIconURL != "" {
+			iconURL = resultIconURL
+		}
+
+		ctx := tomb.Context(nil) // XXX: should this be a real context?
```

Comment:
you mean for e.g. `context.Background()` as a parent?

AFAIU we want the 'tomb' wrapped context as it allows us to detect someone calling Kill() on the associated tomb, and thus allows for the download to be interrupted.

Reply:
Yes, I was thinking `context.Background()`. But it previously had `nil` passed as the `ctx` parameter directly, so I'm not sure what's correct. Would this work?

```suggestion
		ctx := tomb.Context(context.Background())
```